runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SplittableParDo.java

```diff
@@ -117,13 +139,22 @@ public PCollectionTuple expand(PCollection<InputT> input) {
     return keyedRestrictions.apply(
         "ProcessKeyedElements",
         new ProcessKeyedElements<>(
-            fn,
+            doFn,
             input.getCoder(),
             restrictionCoder,
             (WindowingStrategy<InputT, ?>) input.getWindowingStrategy(),
-            parDo.getSideInputs(),
-            parDo.getMainOutputTag(),
-            parDo.getAdditionalOutputTags()));
+            sideInputs,
+            mainOutputTag,
+            additionalOutputTags));
+  }
+
+  @Override
+  public Map<TupleTag<?>, PValue> getAdditionalInputs() {
```

Comment:
These lines are well below my threshold for considering Java code "duplicated" :-)

I've moved the logic into `PCollectionViews.toAdditionalInputs` and invoked it twice here. I also moved some other transforms to use it, in separate commits added to this PR.

To maintain causal commit ordering, I've rebased the utility to the bottom.